lib/search_web/controllers/page_controller.ex

```diff
@@ -1,13 +1,30 @@
 defmodule SearchWeb.PageController do
   use SearchWeb, :controller
 
+  @empty_form Phoenix.Component.to_form(%{
+                "search_text" => nil,
+                "k" => nil,
+                "embedding_model" => nil
+              })
+
+  @embedding_model_opts Search.Application.embedding_models()
+                        |> Enum.map(&(&1 |> Module.split() |> Enum.reverse() |> Enum.at(0)))
+
   def home(conn, _params) do
     # The home page is often custom made,
     # so skip the default app layout.
-    render(conn, :home, form: Phoenix.Component.to_form(%{"search_text" => nil, "k" => nil}))
+    render(conn, :home, form: @empty_form, embedding_models: @embedding_model_opts)
   end
 
-  def search(conn, %{"k" => k, "search_text" => search_text} = params) do
+  def search(
+        conn,
+        %{"k" => k, "search_text" => search_text, "embedding_model" => embedding_model} = params
+      ) do
+    model =
+      Enum.find(Search.Application.embedding_models(), fn mod ->
+        "Elixir.Search.Embeddings.#{embedding_model}" == "#{mod}"
+      end)
```

Comment:
We should decouple the model name from the implementation (module and hierarchy). If we go with the config approach, then we can use the keys, otherwise each module should expose a name or something.